### ISSUE TITLE
Error while running metagenomesimulation.py: "Invalid taxID: ' ' "

### ISSUE THREAD

**Issue by IsaacT1123** [CAMI-challenge/CAMISIM#103]
I'm using the python2.7 version of CAMISIM, and I'm getting the following error while running metagenomesimulation.py: 

`2021-03-22 14:44:09 INFO: [MetagenomeSimulationPipeline] Metagenome simulation starting
2021-03-22 14:44:09 INFO: [MetagenomeSimulationPipeline] Validating Genomes
2021-03-22 14:44:09 INFO: [MetadataReader] Reading file: '[MY PARENT DIRECTORY]/CAMISIM/defaults/new_genome_to_id.tsv'
2021-03-22 14:46:58 INFO: [MetadataReader 9216059885] Reading file: '[MY PARENT DIRECTORY]/CAMISIM/defaults/new_metadata.tsv'
2021-03-22 14:46:58 INFO: [MetadataReader 82380310288] Reading file: '[MY PARENT DIRECTORY]/CAMISIM/out/internal/genome_locations.tsv'
2021-03-22 14:46:59 INFO: [NcbiTaxonomy] Building taxonomy tree...
2021-03-22 14:46:59 INFO: [NcbiTaxonomy] Reading 'nodes' file:  '/tmp/tmpzCnYL8/NCBI/nodes.dmp'
2021-03-22 14:47:19 INFO: [NcbiTaxonomy] Reading 'names' file:  '/tmp/tmpzCnYL8/NCBI/names.dmp'
2021-03-22 14:47:22 INFO: [NcbiTaxonomy] Reading 'merged' file: '/tmp/tmpzCnYL8/NCBI/merged.dmp'
2021-03-22 14:47:23 INFO: [NcbiTaxonomy] Done (23.0s)
2021-03-22 14:47:23 ERROR: [NcbiTaxonomy] Invalid taxid: ''
2021-03-22 14:47:23 ERROR: [MetagenomeSimulationPipeline] Invalid taxid in line 80
2021-03-22 14:47:23 INFO: [MetagenomeSimulationPipeline] Metagenome simulation aborted`

I've deliberately input genomes that are present in the taxdump that CAMISIM is using, so I find this issue quite confusing.
I've attached my config file as well.

Thanks for your help!

[new_miniconfig.txt](https://github.com/CAMI-challenge/CAMISIM/files/6184739/new_miniconfig.txt)

**Comment by AlphaSquad** [comment]
That looks like a taxonomy ID in your `metadata.tsv` file isn't set, could you maybe also check/send this file?

**Comment by IsaacT1123** [comment]
Sure - I've attached it here. it is quite long because I've input a lot of genomes.

[new_metadata.txt](https://github.com/CAMI-challenge/CAMISIM/files/6185823/new_metadata.txt)

**Comment by AlphaSquad** [comment]
Hm, I cannot immediately see something wrong with this. I see that you added a different taxonomy version to your config file. Does your taxonomy have the same structure as the provided taxonomy tarballs? In particular, it needs a head directory called `NCBI` and then the `.dmp` files below that (at least `names.dmp`, `nodes.dmp`, `merged.dmp`). If that is already the case, then I would also need you tax dump to be able to reproduce - I am sorry that it is such a hassle.

**Comment by IsaacT1123** [comment]
No worries - the taxonomy dump is too big to attach here, but I could email it to you instead - what's a good email address I could send this to?
Thanks!

**Comment by AlphaSquad** [comment]
Oh, sure, just send it to adrian.fritz at helmholtz-hzi.de

**Comment by AlphaSquad** [comment]
Alright, I took a deeper dive to find out what's going on and the problem is that there is a few genomes which do not have a complete lineage, e.g. genome75 from your file is `Victivallales bacterium CCUG 44730` and does not have a `family` tax ID in the NCBI, because it is a `unclassified Victivallales`. 
I added a check for missing ranks now, so the simulation should work again and print a warning in the log. The taxonomic profiles will miss these specific genomes at these ranks then, it becomes possible that more sequences are classified at a lower level than at a higher level in the profil, i.e. the Victivallales bacterium species has some abundance, but not its family (since there is none).
